### DISCUSSION TITLE
How to define different keybindings for the same hotkey inside/outside NeoVim?

### DISCUSSION THREAD

**Discussion by ansidev** [ghostty-org/ghostty#4206] [Q&A] [answered]
Story
I want to define the key bindings which match the below behaviour:

| Hotkey | Neovim | Non-neovim |
|--------|--------|--------|
| `Option` + `LeftArrow` | Default Ghostty key binding config: `esc:b` | `Alt` + `LeftArrow` |
| `Option` + `RightArrow` | Default Ghostty key binding config: `esc:f` | `Alt` + `RightArrow` |

How to do that? Are those behaviours supported by Ghostty?

**Comment by rhodes-b** [comment, marked as answer]
this is currently not possible ghostty plans on having a programatic way of controlling things in the future

https://github.com/ghostty-org/ghostty/discussions/2353
the discussion above would be part of the work required to get something like this (and shows it is planned)

**Comment by edrozenberg** [comment]
What about adding bindings in the nvim config (`~/.config/nvim/`) for these keys, and let nvim itself change/override the behavior of these keys while using nvim, rather than needing the terminal to handle this, would it work. I'm mostly a vim user, for vim there is this mapping guide - https://vim.fandom.com/wiki/Mapping_keys_in_Vim_-_Tutorial_(Part_1)

In Ghostty config you'd only need `macos-option-as-alt = true`.

**Reply by rhodes-b** [reply]
I dont think that will work ghostty will steal the keybinds and not pass them through

**Reply by ansidev** [reply]
It doesn't work, that's why I asked, just to verify my assumption.

**Comment by anhoder** [comment]
Should we add a new trigger prefix for this feature?

For example: `when-var:IN_EDITOR=1` would make the keybinding active only when the environment variable `IN_EDITOR` equals 1. (I’m not sure about the naming convention for trigger prefixes; this is just a simple illustration.)

With this, we could toggle different keybindings when starting or exiting Neovim by changing the value of `IN_EDITOR`.

**Reply by rhodes-b** [reply]
ghostty doesnt know about the environment variables of the child process it runs (ie the shell). Environment variables also also definitely a hack almost any other solution would be better.

the config file has historically has been very against control flow type syntax with mitchell initially saying that it would never exist so that is another reason it wouldn't be supported

something like https://github.com/ghostty-org/ghostty/discussions/9902 is a much better way to get different keybinds in neovim (or almost all tuis) as it runs in the alternate buffer (most run in the alt buffer)
